configure.ac

```diff
@@ -881,43 +871,44 @@ if test x$enable_gcc_warnings != xno && test "$GCC" = "yes"; then
   CFLAGS="$CFLAGS -Wno-unused-parameter -Wstrict-aliasing"
 
   if test x$have_gcc4 = xyes ; then
-    # These warnings break gcc 3.3.5 and work on gcc 4.0.2
+    dnl These warnings break gcc 3.3.5 and work on gcc 4.0.2
     CFLAGS="$CFLAGS -Winit-self -Wmissing-field-initializers -Wdeclaration-after-statement"
-    #CFLAGS="$CFLAGS -Wold-style-definition"
+    dnl CFLAGS="$CFLAGS -Wold-style-definition"
   fi
 
   if test x$have_gcc42 = xyes ; then
-    # These warnings break gcc 4.0.2 and work on gcc 4.2
+    dnl These warnings break gcc 4.0.2 and work on gcc 4.2
     CFLAGS="$CFLAGS -Waddress"
   fi
 
   if test x$have_gcc42 = xyes && test x$have_clang = xno; then
-    # These warnings break gcc 4.0.2 and clang, but work on gcc 4.2
+    dnl These warnings break gcc 4.0.2 and clang, but work on gcc 4.2
     CFLAGS="$CFLAGS -Wnormalized=id -Woverride-init"
   fi
 
   if test x$have_gcc45 = xyes ; then
-    # These warnings work on gcc 4.5
+    dnl These warnings work on gcc 4.5
     CFLAGS="$CFLAGS -Wlogical-op"
   fi
 
   if test x$have_clang = xyes; then
-    # Disable the unused-function warnings, because these trigger
-    # for minheap-internal.h related code.
+    dnl Disable unused-function warnings. These trigger for minheap-internal.h.
     CFLAGS="$CFLAGS -Wno-unused-function"
 
-    # clang on macosx emits warnings for each directory specified which
-    # isn't "used" generating a lot of build noise (typically 3 warnings
-    # per file
     case "$host_os" in
         darwin*)
+            dnl Clang on macOS emits warnings for each directory specified which
+            dnl isn't "used", generating a lot of build noise.
             CFLAGS="$CFLAGS -Qunused-arguments"
+            dnl macOS Lion started deprecating the system OpenSSL. Let's just
+            dnl disable all deprecation warnings on macOS; but do so only for GCC.
+            CFLAGS="$CFLAGS -Wno-deprecated-declarations"
```

Comment:
I think it would be reasonable to omit this `-Wno-deprecated-declarations` from CFLAGS in all cases - this is talking about when OS X 10.7, up to OS X 10.13 (ish? maybe?) which had openssl 0.9.8 just for backwards compatibility, but wanted everyone to move to "SecureTransport" or "CoreCrypto". Since macOS 10.13.5 it includes libressl, which is a lot newer and better, but it does not include headers to compile with.

So, I think that if the user ends up trying to use the macOS included libssl/libcrypto in any case today, either it won't work at all, or they deserve any warnings they get. And this CFLAG can be simplified away.

Reply:
Agree, can you make a PR?